tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,286 @@
+      throw 1;
+   }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case of command line parsing arguments
+///
+/// \param[in] output name of the output file
+/// \param[in] inputFiles name of the input .root files
+/// \param[in] name name of the tree
+/// \param[in] expressions what is showed in the histograms
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& output, std::vector<std::string> inputFiles,
+                                 const std::string& name, std::vector<std::string> expressions)
+{
+   fOutputFile=output;
+   fTreeName=name;
+   for (std::string input: inputFiles)
+      fInputFiles.push_back(input);
+   for (std::string expr: expressions) {
+      std::size_t equal=expr.find("=");
+      if (equal == std::string::npos) {
+         ::Error("TSimpleAnalysis",
+                 "Missing '=' in fExpressions in %s",expr.c_str());
+         throw 1;
+      }
+      std::size_t cutPos=expr.find(kCutIntr, equal);
+      std::string substring=expr.substr(0,equal);
+      if (substring.empty()) {
+         ::Error("TSimpleAnalysis",
+                 "No hname found in %s",expr.c_str());
+         throw 2;
+      }
+      fHNames.push_back(substring);
+      substring=expr.substr(equal+1,cutPos-equal-1);
+      if (substring.empty()) {
+         ::Error("TSimpleAnalysis",
+                 "No expression found in %s",expr.c_str());
+         throw 3;
+      }
+      fExpressions.push_back(substring);
+      if (cutPos == std::string::npos) {
+         fCut.push_back("");
+      } else {
+         fCut.push_back(expr.substr(cutPos+kCutIntr.size()));
+      }
+   }
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// Tell if the name of some histograms that will be created are the same
+///
+/// param[in] name name of the histograms
+
+void TSimpleAnalysis::CheckHNames(std::vector<std::string> name)
+{
+   Int_t err=0;
+   for (unsigned i=0; i<name.size()-1; i++)
+      for (unsigned j=i+1; j<name.size(); j++) {
+         if (name[i] == name[j]) {
+            ::Error("TSimpleAnalysis::check_name",
+                    "Multiple %s in the names of histograms (position %d and %d)",
+                    name[i].c_str(), i, j);
+            err++;
+         }
+      }
+   if (err != 0)
+      throw 1;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Return true if the input arguments create the output file correctly
+
+Bool_t TSimpleAnalysis::Analysis()
+{
+   TChain chain(fTreeName.c_str());
+   for (const std::string& inputfile: fInputFiles)
+      chain.Add(inputfile.c_str());
+   TFile ofile(fOutputFile.c_str(),"RECREATE");
+
+   CheckHNames(fHNames);
+
+   for (unsigned i=0; i<fExpressions.size(); i++) {
+      chain.Draw((fExpressions[i] + ">>" + fHNames[i]).c_str(),fCut[i].c_str(),"goff");
+      TH1F *histo = (TH1F*)gDirectory->Get(fHNames[i].c_str());
+      histo->Write();
```

Comment:
What if `histo` is `nullptr` e.g. because the expression was wrong? Could you check for that case and return `false` if so?